src/main/java/org/kairosdb/core/aggregator/FirstAggregator.java

```diff
@@ -17,65 +17,73 @@
 
 import com.google.inject.Inject;
 import org.kairosdb.core.DataPoint;
+import org.kairosdb.core.KairosDataPointFactory;
 import org.kairosdb.core.annotation.FeatureComponent;
-import org.kairosdb.core.datapoints.DoubleDataPointFactory;
+import org.kairosdb.util.KDataInput;
+import org.kairosdb.util.KDataOutput;
 
+import java.io.DataInput;
+import java.io.IOException;
 import java.util.Collections;
 import java.util.Iterator;
 
 /**
- Converts all longs to double. This will cause a loss of precision for very large long values.
+ * Converts all longs to double. This will cause a loss of precision for very large long values.
  */
 @FeatureComponent(
         name = "first",
-		description = "Returns the first value data point for the time range."
+        description = "Returns the first value data point for the time range."
 )
-public class FirstAggregator extends RangeAggregator
-{
-	private DoubleDataPointFactory m_dataPointFactory;
+public class FirstAggregator extends RangeAggregator {
+    private KairosDataPointFactory m_dataPointFactory;
 
-	@Inject
-	public FirstAggregator(DoubleDataPointFactory dataPointFactory)
-	{
-		m_dataPointFactory = dataPointFactory;
-	}
+    @Inject
+    public FirstAggregator(KairosDataPointFactory dataPointFactory) {
```

Comment:
This change to the constructor breaks compilation of the unit tests. It wouldn't be bad to have a test case for this stuff, but I think that at the very least the tests should still be runnable after this change.